composer/utils/file_helpers.py

```diff
@@ -52,6 +52,23 @@
 ]
 
 
+def extract_path_from_symlink(
+    source_path: str,
+    object_store: Optional[Union[ObjectStore, LoggerDestination]] = None,
+) -> str:
+    with tempfile.TemporaryDirectory() as tmpdir:
+        if object_store is not None and isinstance(object_store, ObjectStore):
```

Comment:
I _think_ you don't want the `isinstance(object_store, ObjectStore)` condition here? shouldn't RUD work here too? The type indicates `LoggerDestination` is valid input as the `object_store`

Reply:
We should just convert RUD to RD

Reply:
yeah it RUD could technically work, but only it is never a RUD here unless the user passed the RUD as `load_object_store` to the `Trainer`, which should be deprecated, but. I guess we gotta support it :(

Reply:
If you can find me someone that uses `load_object_store` instead of `load_path`, I'll eat my hat

Reply:
> We should just convert RUD to RD

you mean RU?

Reply:
> yeah it RUD could technically work, but only it is never a RUD here unless the user passed the RUD as `load_object_store` to the `Trainer`, which should be deprecated, but. I guess we gotta support it :(

WandBLogger is actually the use case we really need to make sure we support `LoggerDestination` for

Reply:
resolved